src/app/dashboard/components/add-line-dialog.component.ts

```diff
@@ -23,11 +24,75 @@ import { LineType } from '../types';
   ]
 })
 export class AddLineDialogComponent {
+  #userService: IUserService = inject(UserService);
+
   type: LineType | undefined;
-  types: LineType[] = ['CountStatus', 'Applications', 'Sessions', 'Tasks', 'Partitions', 'Results'];
   typeLabel = $localize`Type`;
   validType: boolean;
 
+  get types(): LineType[] {
+    const allTypes: LineType[] = ['CountStatus', 'Applications', 'Sessions', 'Tasks', 'Partitions', 'Results'];
+    const filteredTypes = allTypes.filter(type => {
+      switch (type) {
+      case 'Sessions':
+        return this.hasSessionsPermission;
+      case 'Tasks':
+        return this.hasTasksPermission;
+      case 'Results':
+        return this.hasResultsPermission;
+      case 'Applications':
+        return this.hasApplicationsPermission;
+      case 'Partitions':
+        return this.hasPartitionsPermission;
+      case 'CountStatus':
+        return this.hasCountStatusPermission;
+      default:
+        return true;
+      }
+    });
+    return filteredTypes;
+  }
+
+  private getUserService(): IUserService {
+    return this.#userService;
+  }
+
+  get hasSessionsPermission(): boolean {
+    const userService = this.getUserService();
+    const hasListSessions = userService.hasPermission('Sessions:ListSessions');
+    const hasGetSession = userService.hasPermission('Sessions:GetSession');
+    return hasListSessions || hasGetSession;
+  }
+
+  get hasTasksPermission(): boolean {
+    const userService = this.getUserService();
+    return userService.hasPermission('Tasks:ListTasks') || userService.hasPermission('Tasks:GetTask');
+  }
+
+  get hasResultsPermission(): boolean {
+    const userService = this.getUserService();
+    return userService.hasPermission('Results:ListResults') || userService.hasPermission('Results:GetResult');
+  }
+
+  get hasApplicationsPermission(): boolean {
+    const userService = this.getUserService();
+    return userService.hasPermission('Applications:ListApplications') || userService.hasPermission('Applications:GetApplication');
+  }
+
+  get hasPartitionsPermission(): boolean {
+    const userService = this.getUserService();
+    return userService.hasPermission('Partitions:ListPartitions') || userService.hasPermission('Partitions:GetPartition');
+  }
+
+  get hasCountStatusPermission(): boolean {
+    const userService = this.getUserService();
+    return userService.hasPermission('Tasks:CountTasksByStatus');
+  }
+
+  get hasAvailableTypes(): boolean {
+    return this.types.length > 0;
+  }
+
```

Comment:
Maybe we could add a method in the user service that would check a string of permissions.